package/test/index.test.ts

```diff
@@ -251,4 +293,103 @@ describe('requestHandler', () => {
     const prices: number[] = hits.map((hit: { price: number }) => hit.price);
     expect([...prices]).toEqual(prices.sort((a, b) => b - a));
   });
+
+  it('should handle returning columns', async () => {
+    // Prerequisites
+    await initTestDatabase(initTestDatabaseParams);
+    await createColumnAndIndex({ tableName });
+
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn().mockReturnThis(),
+    };
+
+    const req = {
+      body: {
+        params: { query: 'football' },
+        indexName: tableName,
+        pgOptions: { returnColumns: ['name', 'description'] },
+      },
+    };
+    await searchHandler(req, res, [
+      { tableName, validReturnColumns: ['name', 'description'] },
+    ]);
+
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalled();
+
+    const hits = res.json.mock.calls[0][0].results[0].hits;
+
+    // All hist should only have properties name and description
+    hits.forEach((hit: any) => {
+      expect(Object.keys(hit)).toEqual(['name', 'description']);
+    });
+
+    /**
+     * Invalid columns
+     */
+
+    await searchHandler(req, res, [
+      { tableName, validReturnColumns: ['name'] },
+    ]);
+
+    // It should have returned an error
+    expect(res.status).toHaveBeenCalledWith(400);
+    expect(res.json).toHaveBeenCalledWith({
+      error: 'Request contained invalid payload',
+    });
+  });
+
+  it('should handle highlighting', async () => {
+    // Prerequisites
+    await initTestDatabase(initTestDatabaseParams);
+    await createColumnAndIndex({ tableName });
+
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn().mockReturnThis(),
+    };
+
+    const req = {
+      body: {
+        params: {
+          query: 'awesome football',
+          page: 0,
+          highlightPreTag: '__ais-highlight__',
+          highlightPostTag: '__/ais-highlight__',
+        },
+        indexName: tableName,
+        pgOptions: {
+          highlightColumns: ['name', 'description'],
+        },
+      },
+    };
+    await searchHandler(req, res, [
+      { tableName, validHighlightColumns: ['name', 'description'] },
+    ]);
+
+    // Test status and json are called
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.json).toHaveBeenCalled();
+
+    const hits = res.json.mock.calls[0][0].results[0].hits;
+
+    // It should have highlighted both of the query terms
+    // in the name and description
+    expect(hits).toMatchSnapshot();
```

Comment:
This is the check for the highlight string in both the name and description. re. review quesiton